src/vivarium/framework/results/observation.py

```diff
@@ -35,6 +35,9 @@
 
 VALUE_COLUMN = "value"
 
+if TYPE_CHECKING:
+    _PandasGroup = pd.DataFrame | DataFrameGroupBy[tuple[str, ...] | str]
```

Comment:
Figures this isn't used anywhere else so just made it private here instead of adding to types.py